ui/src/pages/Recover.js

```diff
@@ -0,0 +1,275 @@
+import { useEffect, useState } from "react"
+import { Button, Input, Label, Form, Dropdown } from 'semantic-ui-react';
+import "../App.css";
+import { appWindow, LogicalSize } from '@tauri-apps/api/window';
+import Loading from './Loading';
+import HyperLinkButton from "../components/HyperLinkButton";
+
+const bip39 = require('bip39');
+
+const SEED_PHRASE_PAGE = 0;
+const NEW_PASSWORD_PAGE = 1;
+const FINISH_PAGE = 2;
+const LOADING_TAB = 3;
+
+const MIN_PASSWORD_LENGTH = 8;
+
+const DEFAULT_WINDOW_SIZE = new LogicalSize(460, 500);
+const CONFIRM_PHRASE_WINDOW_SIZE = new LogicalSize(460, 750);
+
+const DROPDOWN_OPTIONS = [
+  {
+    text: "I have a 12 word phrase",
+    value: 12
+  },
+  {
+    text: "I have a 24 word phrase",
+    value: 24
+  },
+];
+
+const DEFAULT_PHRASES = {
+  12: ["", "", "", "", "", "", "", "", "", "", "", ""],
+  24: ["", "", "", "", "", "", "", "", "", "", "", "",
+    "", "", "", "", "", "", "", "", "", "", "", ""]
+}
```

Comment:
18 word phrases are also possible. Even if we don't ever allow users to create 18 word phrases, it's part of the BIP spec, so we should still support it.